### PR TITLE
fix issue that amp bf16 does not work for cpu in env with cuda.

### PR DESCRIPTION
**Pull request by sywangyi**
and num_cpu_threads_per_process is not reset for better performance in cpu only case

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

### CODE REVIEW

**Comment by sywangyi** [comment]
@sgugger @yao-matrix  @jiqing-feng  please notice the issue

File: src/accelerate/accelerator.py [hunk 1]
```diff
@@ -365,7 +365,10 @@ def __init__(
             and self.distributed_type != DistributedType.FSDP
             and self.distributed_type != DistributedType.MEGATRON_LM
         ):
-            self.native_amp = is_bf16_available(True)
+            if self.device.type == "cpu":
+                self.native_amp = is_torch_version(">=", "1.10")
+            else:
+                self.native_amp = is_bf16_available(True)
```
Comment (muellerzr) [inline src/accelerate/accelerator.py:371]: I don't understand the need for this, `is_bf16_available` already does this check as shown here: https://github.com/huggingface/accelerate/blob/6ebaab71bbf87960583bc17339c9da9ec2e4ea91/src/accelerate/utils/imports.py#L89-L92

File: src/accelerate/accelerator.py [hunk 2]
```diff
@@ -948,7 +951,7 @@ def prepare_model(self, model: torch.nn.Module, device_placement=None):
             if self.mixed_precision == "fp16" and is_torch_version(">=", "1.10"):
                 model.forward = torch.cuda.amp.autocast(dtype=torch.float16)(model.forward)
             elif self.mixed_precision == "bf16" and self.distributed_type != DistributedType.TPU:
-                device_type = "cuda" if torch.cuda.is_available() else "cpu"
+                device_type = "cuda" if self.device.type != "cpu" and torch.cuda.is_available() else "cpu"
```
Comment (muellerzr) [inline src/accelerate/accelerator.py:954]: This line (and the next occurrence) can be removed entirely and just pass in `self.device.type` to `torch.autocast`

**Review by muellerzr — COMMENTED** [review]
Thanks for the enhancement! Rather than having an ever-growing conditional, I think we should just pass `self.device.type` to `autocast` instead directly to avoid this, since it is already setup based on what was done in `accelerate.prepare`

File: src/accelerate/accelerator.py (hunk 2 above)
Comment (sywangyi) [reply src/accelerate/accelerator.py]: good point. done

Comment (sywangyi) [reply src/accelerate/accelerator.py]: done

**Review by muellerzr — APPROVED** [review]

**Pull request merged**